// jozsef-lib/directive/expandableRepeat.js
function expandableRepeat() {
  function ctrl($scope, $element, $compile) {
    const repeatStr = $($element).attr('expandable-repeat');
    const split = repeatStr.replace(/\s{1,}/g, ' ').split(' in ');
    const item = split[0];
    const array = split[1];
    if ($scope[array] === undefined) {
      $scope[array] = [{}];
    }
    const repeatTemplate = $($element).html();

    function add() {
      $scope[array].push({});
    }

    function remove(index) {
      $scope[array].splice(index, 1);
    }

    $scope.add = add;
    $scope.remove = remove;

    const template =
      `<div>
        <div ng-repeat='${repeatStr}'>
          ${repeatTemplate}
          <button class='button close-btn' ng-click='remove($index)'>X</button>
        </div>
        <br>
        <div class='center'>
          <button tabindex="0" class='button' ng-click='add()'>Add ${item}</button>
        </div>
      </div>`

    $($element).html($compile(template)($scope));
  }
  return {
    restrict: 'A',
    controller: ctrl,
  };
}

exports.expandableRepeat = expandableRepeat;
